### PR TITLE
added get_semantic_model_size

### PR DESCRIPTION
**Pull request by m-kovalsky**

### CODE REVIEW

File: src/sempy_labs/_generate_semantic_model.py [hunk 1]
```diff
@@ -257,3 +257,35 @@ def get_semantic_model_bim(
...
+    )
+
+    used_size = fabric.evaluate_dax(
+        dataset=dataset,
+        workspace=workspace,
+        dax_string="""
+        EVALUATE SELECTCOLUMNS(INFO.STORAGETABLECOLUMNSEGMENTS(),[USED_SIZE])
+        """,
+    )
+    dict_size = dict["[DICTIONARY_SIZE]"].sum()
```
Comment (eisber) [inline src/sempy_labs/_generate_semantic_model.py:281]: can't you SUM in DAX?

File: src/sempy_labs/_list_functions.py [hunk 2]
```diff
@@ -97,25 +99,146 @@ def list_tables(dataset: str, workspace: Optional[str] = None) -> pd.DataFrame:
...
+    ) as tom:
+        if extended:
+            dict_df = fabric.evaluate_dax(
+                dataset=dataset,
+                workspace=workspace,
+                dax_string="""
+                EVALUATE SELECTCOLUMNS(FILTER(INFO.STORAGETABLECOLUMNS(), [COLUMN_TYPE] = "BASIC_DATA"),[DIMENSION_NAME],[DICTIONARY_SIZE])
+                """,
+            )
+            dict_sum = dict_df.groupby("[DIMENSION_NAME]")["[DICTIONARY_SIZE]"].sum()
```
Comment (eisber) [inline src/sempy_labs/_list_functions.py:138]: why not SUM on the DAX side?

File: src/sempy_labs/_list_functions.py [hunk 3]
```diff
@@ -97,25 +99,146 @@ def list_tables(dataset: str, workspace: Optional[str] = None) -> pd.DataFrame:
...
+                        "User Hierarchy Size": u_size,
+                    }
+                )
+
+            rows.append(new_data)
+
+        df = pd.DataFrame(rows)
+
+        if extended:
+            int_cols = [
```
Comment (eisber) [inline src/sempy_labs/_list_functions.py:232]: maybe it's easier to just do "Total Size": int(total_size)?

File: src/sempy_labs/_generate_semantic_model.py (hunk 1 above)
Comment (m-kovalsky) [reply src/sempy_labs/_generate_semantic_model.py:281]: I feel a bit dumb here but I'm not sure of a way to sum it up in this context when the column in question is not actually a column in the model.

File: src/sempy_labs/_list_functions.py (hunk 2 above)
Comment (m-kovalsky) [reply src/sempy_labs/_list_functions.py:138]: same

File: src/sempy_labs/_list_functions.py (hunk 3 above)
Comment (m-kovalsky) [reply src/sempy_labs/_list_functions.py]: fixed

File: src/sempy_labs/_generate_semantic_model.py [hunk 4]
```diff
@@ -257,3 +257,35 @@ def get_semantic_model_bim(
...
+    used_size = used_size["[USED_SIZE]"].sum()
+    model_size = dict_size + used_size
+    if model_size >= 10**9:
+        result = model_size / (1024**3) * 10**9
```
Comment (eisber) [inline src/sempy_labs/_generate_semantic_model.py:285]: can you add a comment on what's going on here?

**Review by eisber — APPROVED** [review]

**Pull request merged**